### PR TITLE
net-p2p/classified-ads: Classified ads v0.10 version bump.

### PR DESCRIPTION
**Pull request by operatornormal**
Here is new ebuild for classified ads version 0.10. Repoman is happy.

### CODE REVIEW

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	cd test || die "test suite missing"
+	"$(qt4_get_bindir)"/qmake || die "test suite configure failed"
+	emake
+	# test suite will create files under $HOME, set $HOME to point to
+	# safe location, ideas stolen from
+	# eclass/distutils-r1.eclass func distutils_install_for_testing
+	BACKUP_HOME=$HOME
+	export HOME=${BUILD_DIR}/tmp
+	mkdir -p $HOME || true
+	./testca
```
Comment (monsieurp) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: What is this script meant to be doing?

Comment (monsieurp) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: Run the test suite. Alright. What you might be able to do here is this:
```./testca || die "error during tests"```. But I'm not happy with the way this is run.

**Comment by monsieurp** [comment]
Hello and thank you for the PR. It needs some improvements though. @idella will take it from there since I see his name listed in the metadata.xml file.

**Comment by rindeal** [comment]
Lenin on the homepage is intended?

File: net-p2p/classified-ads/classified-ads-0.10.ebuild (hunk 1 above)
Comment (idella) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: monsieurp not this time. Sometimes authors write a shell script and to run it you run  /path/to/script.sh
In this case it lacks the .sh suffix.  Use of || die pertains to executables that are system installed, though could be applied to such custom scripts.  Any such script should be equipped with its own internal from of abort on fail returning a non 0 value.

**Comment by operatornormal** [comment]
Thank you for your comments,

 - The way how test suite is run has its reasons and it raises questions every time there is a change in this ebuild, see for example https://github.com/gentoo/gentoo/pull/500 for previous round. Making the test-suite part look better would require changes in method how files created during test phase are cleaned away
 - Lenin checking out classified ads section of Правда (the truth) is very much intentional :)

So, is there anything that really should be changed, except for future development item regarding the test suite? 

--
Antti Järvinen

**Comment by idella** [comment]
One thing you do need do is bump to EAPI 6

I cannot add it yet due to this

```bash
QDEBUG : TestClassifiedAds::tryDeletingController() "18:39:04" Voice call engine mockup is no more 
PASS   : TestClassifiedAds::tryDeletingController()
PASS   : TestClassifiedAds::cleanupTestCase()
Totals: 31 passed, 2 failed, 0 skipped
********* Finished testing of TestClassifiedAds *********
 * ERROR: net-p2p/classified-ads-0.10::gentoo failed (test phase):
 *   test suite failed with error code  2
 * 
```

**Comment by operatornormal** [comment]
EAPI is now 6, program depends on qt5 instead of qt4. Version number remains the same 0.10 but source needs to be re-fetched from upstream because 2 unit tests were fixed in the source.

**Comment by idella** [comment]
On re-testing still same result.  This is still waiting to be coded into the ebuild ?

**Comment by idella** [comment]
The fail resulted form an issue with inconsistency in the state of compiled deps form gcc, a known issue. Passes all now.

**Comment by operatornormal** [comment]
Tried to --force the Manifest, is the digest now ok? 

--
Antti

**Comment by Polynomial-C** [comment]
Reopening as these commits have been reverted in Gentoo's portage tree.

Would it be possible to merge all four commits into one? From what I see this is a new ebuild and thus should be added in one commit.

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+KEYWORDS="~amd64 ~x86"
+
+IUSE="debug doc test"
+
+RDEPEND="dev-libs/openssl:0
+		>=net-libs/libnatpmp-20130911
+		<=net-libs/libnatpmp-20140401-r1
+		>=net-libs/miniupnpc-1.8
+		sys-apps/file
+		sys-devel/gettext
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: gettext is usually not required in RDEPEND but only in DEPEND. Please check carefully if it really needs to be in RDEPEND.

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+# virtualx requires a command that returns number, and does not just die:
+test_suite() {
+	cd test || return -1
+	eqmake5 || return -2
+	emake
+	# test suite will create files under $HOME, set $HOME to point to
+	# safe location, ideas stolen from
+	# eclass/distutils-r1.eclass func distutils_install_for_testing
+	BACKUP_HOME=$HOME
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: Please change that to ```BACKUP_HOME="${HOME}"```

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	export HOME=${BUILD_DIR}/tmp
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: likewise ```export HOME="${BUILD_DIR}/tmp"```

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	mkdir -p $HOME || true
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: Same here: ```mkdir -p "${HOME}" || true```

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	./testca
+	result=$?
+	export HOME=$BACKUP_HOME
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: Again... quoting + curly braces: ```export HOME="${BACKUP_HOME}"```

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+	if [ $result != "0" ]; then
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: Please use bash test syntax:
```if [[ "${result}" -ne 0 ]] ; then```

File: net-p2p/classified-ads/classified-ads-0.10.ebuild [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+		echo "test suite failed with error code " `echo $result`
+		return $result
```
Comment (Polynomial-C) [inline net-p2p/classified-ads/classified-ads-0.10.ebuild]: quoting + curly braces again...

**Comment by idella** [comment]
As a general comment, the use of quoting + curly braces again is the more secure and bullet proof code syntax. On the other hand, the ebuild works as is without them and passes all demands of it in runtesting.  If there is any residual doubt I have it is over the SRC_URI.  
Is this form in the ebuild going to safeguard against any other possible 'illegal' substitution of the tarball under the same name? Can the repo source be set to survive and such event without having a conflicting set of hash values in the Manifest?

The issues with the Manifest were also confounded by the github mirror system not managing them over 4 commits with forced repushes and a requirement of full --rebase.
Overall I suggest to take all the corrections, close this pull req and  make a fresh one with all edited correct ebuild and Manifest. The final state of this pull request did not survive all the changes well

**Comment by gktrk** [comment]
I think overriding HOME violates the PMS. According to [Table 11.1](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-119002), HOME is a read-only variable. Unfortunately, repoman [does not check](https://github.com/gentoo/portage/blob/master/pym/repoman/modules/scan/ebuild/checks.py#L239) for it at the moment. Nonetheless, the PMS says: "The full path to an appropriate temporary directory for use by any programs invoked by the ebuild that may read or modify the home directory." So leaving HOME as is should be fine. Any thoughts?

**Comment by operatornormal** [comment]
All right, I'll create a new PR that fixes the QA issues listed above and contains only one (clean) version of the ebuild.  This one doesn't need to be handled any more. 

--
Antti

**Pull request closed without merging**